src/pool/cli.c

```diff
@@ -1763,12 +1798,16 @@ map_refresh_cb(tse_task_t *task, void *varg)
 		pool->dp_map_sz = out->tmo_map_buf_size;
 		reinit = true;
 		goto out;
-	} else if (rc == -DER_AGAIN) {
+	} else if (daos_rpc_retryable_rc(rc) || rc == -DER_AGAIN) {
+		D_DEBUG(DB_MD, DF_UUID": %p: retryable: "DF_RC"\n", DP_UUID(pool->dp_pool), task,
+			DP_RC(rc));
 		reinit = true;
 		goto out;
 	} else if (rc != 0) {
-		D_ERROR(DF_UUID": failed to fetch pool map: "DF_RC"\n",
-			DP_UUID(pool->dp_pool), DP_RC(rc));
+		D_DEBUG(DB_MD, DF_UUID": %p: serious: "DF_RC"\n", DP_UUID(pool->dp_pool), task,
+			DP_RC(rc));
+		arg->mra_n--;
+		reinit = true;
```

Comment:
will it finally cause fall through (create new independent pool query task) again and again for some un-retryable error? for example DER_NOMEM or other un-retryable err?

Reply:
If rc == -DER_NOMEM here, we know that (1) it is from the reply buffer and (2) the engine side has encountered a memory allocation failure, and we decrement mra_n and rerun the map_refresh to choose another engine. When choosing another engine, if mra_n has reached 0, we fall back to dc_pool_query, in which case we don't use or run map_refresh_cb anymore.

Reply:
ok, thanks